src/MIDebugEngine/Engine.Impl/ExceptionManager.cs

```diff
@@ -480,17 +525,30 @@ private async Task UpdateCatagory(Guid categoryId, ExceptionCategorySettings cat
                     }
                 }
 
-                IEnumerable<ulong> breakpointIds = await _commandFactory.SetExceptionBreakpoints(categoryId, exceptionNames, grouping.Key);
-
-                int count = exceptionNames.Zip(breakpointIds, (exceptionName, breakpointId) =>
+                bool isBreakThrown = grouping.Key.HasFlag(ExceptionBreakpointStates.BreakThrown);
+                if (!IsCppExceptionsCategoryChecked(updates) && isBreakThrown)
+                {
+                    // IEnumerable<ulong> breakpointIds = await _commandFactory.SetExceptionBreakpoints(categoryId, exceptionNames, grouping.Key);
+                    // this approach is very messy, but we need to update categorySettings.CurrentRules
+                    foreach (string exceptionName in exceptionNames)
+                    {
+                        IEnumerable<ulong> breakpointIds = await _commandFactory.SetExceptionBreakpoints(categoryId, new string[] { exceptionName }, grouping.Key);
+                        categorySettings.CurrentRules.Add(exceptionName, breakpointIds.First());
+                    }
+                }
```

Comment:
>} [](http://example.com/codeflow?start=16&length=1)

We should send a warning to the output window when the category is checked but there is an unchecked exception.

Reply:
I am tempted to keep `IsCppExceptionsCategoryChecked()` as we can use `if (!updates.RulesToAdd[rule].HasFlag(ExceptionBreakpointStates.BreakThrown))` to check for unchecked exceptions when category is checked and send the warning before returning false. Thoughts on this? 

And by sending a warning to the output window, we want to write to console, correct? Would a MessageBox be better as it is more visible? But then again, sending the warning for each unchecked exception in RulesToAdd would be bothersome from a customer's perspective.

```
private bool IsCppExceptionsCategoryChecked(SettingsUpdates updates)
{
    // C++ exceptions category is checked when NewCategoryState is BreakThrown and all RulesToAdd contain BreakThrown
    if (updates.NewCategoryState == ExceptionBreakpointStates.BreakThrown)
    {
        foreach (string rule in updates.RulesToAdd.Keys)
        {
            if (!updates.RulesToAdd[rule].HasFlag(ExceptionBreakpointStates.BreakThrown))
            {
                // WE CAN SEND WARNING HERE
                return false;
            }
        }
        return true;
    }
    return false;
}
```

Reply:
> I am tempted to keep `IsCppExceptionsCategoryChecked()`

You shouldn't need it. You can find if you have exceptions that are unchecked while the category is checked with:

```
else if (grouping.Key != categorySettings.CategoryState && !grouping.Key.HasFlag(ExceptionBreakpointStates.BreakThrown))
{
   // Send warning
}
```

Reply:
> And by sending a warning to the output window, we want to write to console, correct?  Would a MessageBox be better as it is more visible? But then again, sending the warning for each unchecked exception in RulesToAdd would be bothersome from a customer's perspective.

By the console do you mean the program's console? Or the output window? My suggestion was the debug pain of the output window. We don't want to write to the target program's console.

A message box would definitely be more visible, but it would also be more annoying if you have the same solution that you are going back and forth between Windows and Linux and you don't actually even care about stopping on C++ exceptions in Linux and none are raised, so it is a bit of a trade-off.

If there is a way to get extra text to show up when we stop on an exception indicating that we don't support unclicking specific exceptions then that would probably be the best option, and then I would definitely go with just an output window warning for exception configuration time.

If we can't make that happen then maybe we are both right: a message box the first time we see exceptions configured this way in a debug session, and an output window warning and any subsequent times.

Reply:
Strange. I tried using `Debug.WriteLine("insert-string-here");`, but the warning did not show in the Output window...

Reply:
Used `HostOutputWindow.WriteLaunchError("There are unchecked exceptions in a checked exceptions category.");` to resolve this issue. Thanks @caslan.